File: WebClient/src/app/models/registration-details.ts
export class RegistrationDetails {
  FirstName: string;
  LastName: string;
  Email: string;
  Phone: string;
  Address: string;
  Country: string;
  DOB: Date | null;
  Password: string;
  Subscribed: boolean;

  constructor(
      FirstName: string, LastName: string, Email: string,
      Phone: string, Address: string, Country: string,
      DOB: Date | null, Password: string, Subscribed: boolean
  ) {
      this.FirstName = FirstName;
      this.LastName = LastName;
      this.Email = Email;
      this.Phone = Phone;
      this.Address = Address;
      this.Country = Country;
      this.DOB = DOB;
      this.Password = Password;
      this.Subscribed = Subscribed;
  }
}

File: Server/Models/RegistrationDetails.cs
using System;

namespace Server.Models
{
    public class RegistrationDetails
    {
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public string Country {get; set; } = string.Empty;
        public DateTime? DOB {get; set;}
        public string Password { get; set; } = string.Empty;
        public bool Subscribed {get; set; }
    }
}
